[llvm] llvm-tblgen.exe fails when building LLVM

**Host Environment**
- OS: Windows 10 2004
- Compiler: Microsoft (R) C/C++ Optimizing Compiler Version 19.26.28805 for x86

**To Reproduce**
Steps to reproduce the behavior:
`./vcpkg install llvm`

**Failure logs** 
-Cut and past the appropriate build messages from the console output.
```
PS D:\vcpkg2> .\vcpkg install llvm
Computing installation plan...
The following packages will be built and installed:
    llvm[clang,core,disable-abi-breaking-checks,disable-assertions,disable-clang-static-analyzer,enable-rtti,lld,tools]:x86-windows
Starting package 1/1: llvm:x86-windows
Building package llvm[clang,core,disable-abi-breaking-checks,disable-assertions,disable-clang-static-analyzer,enable-rtti,lld,tools]:x86-windows...
-- Note: llvm only supports static library linkage. Building static library.
-- Downloading https://github.com/llvm/llvm-project/archive/llvmorg-10.0.0.tar.gz...
-- Extracting source D:/vcpkg2/downloads/llvm-llvm-project-llvmorg-10.0.0.tar.gz
-- Applying patch 0001-allow-to-use-commas.patch
-- Applying patch 0002-fix-install-paths.patch
-- Using source at D:/vcpkg2/buildtrees/llvm/src/org-10.0.0-0eb85e4539
-- Downloading https://www.python.org/ftp/python/3.7.3/python-3.7.3-embed-win32.zip...
-- Configuring x86-windows
-- Building x86-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed: D:/vcpkg2/downloads/tools/cmake-3.17.2-windows/cmake-3.17.2-win32-x86/bin/cmake.exe --build . --config Debug --target install -- -v
    Working Directory: D:/vcpkg2/buildtrees/llvm/x86-windows-dbg
    See logs for more information:
      D:\vcpkg2\buildtrees\llvm\install-x86-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/llvm/portfile.cmake:198 (vcpkg_install_cmake)
  scripts/ports.cmake:90 (include)


Error: Building package llvm:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: llvm:x86-windows
  Vcpkg version: 2020.02.04-nohash

Additionally, attach any relevant sections from the log files above.
```

-Please attach any additional failure logs mentioned in the console output.
```
[444/2867] cmd.exe /C "cd /D D:\vcpkg\buildtrees\llvm\x86-windows-dbg && D:\vcpkg\buildtrees\llvm\x86-windows-dbg\bin\llvm-tblgen.exe -gen-intrinsic-enums -intrinsic-prefix=mips -I D:/vcpkg/buildtrees/llvm/src/org-10.0.0-0eb85e4539/llvm/include/llvm/IR -I D:/vcpkg/buildtrees/llvm/src/org-10.0.0-0eb85e4539/llvm/include D:/vcpkg/buildtrees/llvm/src/org-10.0.0-0eb85e4539/llvm/include/llvm/IR/Intrinsics.td --write-if-changed -o include/llvm/IR/IntrinsicsMips.h -d include/llvm/IR/IntrinsicsMips.h.d"
FAILED: include/llvm/IR/IntrinsicsMips.h 
cmd.exe /C "cd /D D:\vcpkg\buildtrees\llvm\x86-windows-dbg && D:\vcpkg\buildtrees\llvm\x86-windows-dbg\bin\llvm-tblgen.exe -gen-intrinsic-enums -intrinsic-prefix=mips -I D:/vcpkg/buildtrees/llvm/src/org-10.0.0-0eb85e4539/llvm/include/llvm/IR -I D:/vcpkg/buildtrees/llvm/src/org-10.0.0-0eb85e4539/llvm/include D:/vcpkg/buildtrees/llvm/src/org-10.0.0-0eb85e4539/llvm/include/llvm/IR/Intrinsics.td --write-if-changed -o include/llvm/IR/IntrinsicsMips.h -d include/llvm/IR/IntrinsicsMips.h.d"
llvm-tblgen.exe: Unknown command line argument '-gen-intrinsic-enums'.  Try: 'D:\vcpkg\buildtrees\llvm\x86-windows-dbg\bin\llvm-tblgen.exe --help'
llvm-tblgen.exe: Did you mean '--version'?
llvm-tblgen.exe: Unknown command line argument '-intrinsic-prefix=mips'.  Try: 'D:\vcpkg\buildtrees\llvm\x86-windows-dbg\bin\llvm-tblgen.exe --help'
llvm-tblgen.exe: Did you mean '--info-output-file=mips'?
```

**Additional context**

--- a/ports/llvm/0003-fix-vs2019-v16.6.patch
+++ b/ports/llvm/0003-fix-vs2019-v16.6.patch
@@ -1,0 +1,15 @@
+diff --git a/llvm/include/llvm/Support/ManagedStatic.h b/llvm/include/llvm/Support/ManagedStatic.h
+index bbd0d04ed..f2b41422f 100644
+--- a/llvm/include/llvm/Support/ManagedStatic.h
++++ b/llvm/include/llvm/Support/ManagedStatic.h
+@@ -40,8 +40,8 @@ template <typename T, size_t N> struct object_deleter<T[N]> {
+ // constexpr, a dynamic initializer may be emitted depending on optimization
+ // settings. For the affected versions of MSVC, use the old linker
+ // initialization pattern of not providing a constructor and leaving the fields
+-// uninitialized.
+-#if !defined(_MSC_VER) || defined(__clang__)
++// uninitialized. See http://llvm.org/PR41367 for details.
++#if !defined(_MSC_VER) || (_MSC_VER >= 1925) || defined(__clang__)
+ #define LLVM_USE_CONSTEXPR_CTOR
+ #endif
+ 

--- a/ports/llvm/CONTROL
+++ b/ports/llvm/CONTROL
@@ -1,5 +1,5 @@
 Source: llvm
-Version: 10.0.0-2
+Version: 10.0.0-3
 Homepage: https://llvm.org/
 Description: The LLVM Compiler Infrastructure
 Supports: !uwp

--- a/ports/llvm/portfile.cmake
+++ b/ports/llvm/portfile.cmake
@@ -13,6 +13,7 @@ vcpkg_from_github(
     PATCHES
         0001-allow-to-use-commas.patch
         0002-fix-install-paths.patch
+        0003-fix-vs2019-v16.6.patch
 )
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS

